src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -198,6 +198,26 @@ int sd_dhcp6_client_set_duid(sd_dhcp6_client *client, uint16_t duid_type,
         return 0;
 }
 
+int sd_dhcp6_client_get_duid(sd_dhcp6_client *client, uint16_t *duid_type,
+                             const uint8_t **duid, size_t *duid_len) {
+        assert_return(client, -EINVAL);
+        assert_return(duid_type, -EINVAL);
+        assert_return(duid_len, -EINVAL);
+        assert_return(duid, -EINVAL);
+        assert_return(IN_SET(client->state, DHCP6_STATE_STOPPED), -EBUSY);
+
+        *duid_type = 0;
+        *duid_len = 0;
+        *duid = NULL;
+        if (client->duid_len > 0) {
+                *duid_type = be16toh(client->duid.type);
+                *duid_len = client->duid_len - sizeof(client->duid.type);
+                *duid = client->duid.raw.data;
```

Comment:
This looks error-prone: `xxx_len` should really match `xxx` exactly. It we have to keep adding and dropping some offset we're going to get this wrong sooner or later.